### ISSUE TITLE
Can I use the ProcessLocalImage.ipynb to derive ET of urban scenes based on UAV images?

### ISSUE THREAD

**Issue by HazelSCUT** [hectornieto/pyTSEB#53]
Dear colleagues,

Recently, I have tried to use ProcessLocalImage.ipynb to process my urban data that were derived from UAV. I used the model of Priestley Taylor. However, the results seems to have many abnormal values (i.e. NaN): The black pixels of the following pictures show NaN, and such areas are almost trees. I don't know whether can I use this method to derive the spatial ET of urban scenes.

![sensible heat flux](https://user-images.githubusercontent.com/61641190/226098585-1b1010f7-b79c-418d-a8f2-727be4ac4884.png)
![latent heat flux](https://user-images.githubusercontent.com/61641190/226098593-dd0aaaf0-6bb2-41da-93fd-9e7267ad77bf.png)
![soil heat flux](https://user-images.githubusercontent.com/61641190/226098594-590c3085-5f4d-44ad-b4ca-332d072f9352.png)
![net radiation](https://user-images.githubusercontent.com/61641190/226098596-9f02439a-9d94-4050-a0df-026297b9e29b.png)
![test area](https://user-images.githubusercontent.com/61641190/226098683-e1838f0e-51f6-4d87-a35b-6e489cdf6356.png)

Thanks.

**Comment by HazelSCUT** [comment]
By the way, the pictures respectively are:  Sensible heat flux, Latent heat flux, Soil heat flux, Net radiation and test area.

**Comment by hectornieto** [comment]
Usually one should not apply the model to urban environments, especially in thos very patchy conditinos. The model assumes that pixels are not affected by neiboring conditions, which in the case of urban environmnents, buildins are affecting radiaiton and turbulence (eddies). I.e. a rather homogeneity in the scene and medium scale is expected. Your AOI is extremele heterogeneous and buildings around, but alos tiles and walkpaths are affecting the micrometeorology of the green patches. So I would not advice use this model for your purposes. In any case, I do not think the NaN are caused by this issue. So if you want me to dig deeper in what is causing the problem, please share your configuration data and and example (subset) of the inputs you are using

**Comment by HazelSCUT** [comment]
Hello,


Thanks for your reply. Ok, I understand it. 


However, I want to know why NaN values occur. The subset of my dataset is attached. Other configurations are also attached. Please check it. 


PS: 
1) I did not separately mask each type of vegetation, instead I use the whole image. 
2) As the spatial resolution of my data is 7cm, so the fractional vegetation cover was assumed to be 1.0. I don't know whether it is correct. 
By the way, I have attached the fraction vegetation cover.tif that was calculated based on NDVI.
3) The Green Fraction data (greeneryindex.tif) was calculated based on NDVI. I have also attached NDVIimage.tif that has negative values due to the existence of impermeable surfaces.
4)Other parameters in the configuration were set as default. The configuration file was also attached.




Thanks.


------------------ 原始邮件 ------------------
发件人:                                                                                                                        "hectornieto/pyTSEB"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年3月20日(星期一) 晚上7:14
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [hectornieto/pyTSEB] Can I use the ProcessLocalImage.ipynb to derive ET of urban scenes based on UAV images? (Issue #53)





 
Usually one should not apply the model to urban environments, especially in thos very patchy conditinos. The model assumes that pixels are not affected by neiboring conditions, which in the case of urban environmnents, buildins are affecting radiaiton and turbulence (eddies). I.e. a rather homogeneity in the scene and medium scale is expected. Your AOI is extremele heterogeneous and buildings around, but alos tiles and walkpaths are affecting the micrometeorology of the green patches. So I would not advice use this model for your purposes. In any case, I do not think the NaN are caused by this issue. So if you want me to dig deeper in what is causing the problem, please share your configuration data and and example (subset) of the inputs you are using

**Comment by hectornieto** [comment]
Sorry but I cannot find your configuration file nor your attached images. 

But the effective resolution that the model should be run is at a spatial scale effectively larger than the main vegetation elements. It does not really make sense to estimate fluxes at 7cm resolution since this process happens at a much coarser scale due to radiation transmission efficiency, tubulent transport and plant hydraulics/phisiology. That is the main reason of the roles of fc and fg together with LAI.

All the best

**Comment by HazelSCUT** [comment]
Ok, but why have you also used drone images with high spatial resolution for your study? The difference is between homogeneous farmland and highly heterogeneous urban surfaces rather than the spatial resolution. Thanks.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "hectornieto/pyTSEB"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年3月21日(星期二) 凌晨3:08
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [hectornieto/pyTSEB] Can I use the ProcessLocalImage.ipynb to derive ET of urban scenes based on UAV images? (Issue #53)





 
Sorry but I cannot find your configuration file nor your attached images.
 
But the effective resolution that the model should be run is at a spatial scale effectively larger than the main vegetation elements. It does not really make sense to estimate fluxes at 7cm resolution since this process happens at a much coarser scale due to radiation transmission efficiency, tubulent transport and plant hydraulics/phisiology. That is the main reason of the roles of fc and fg together with LAI.
 
All the best

**Comment by hectornieto** [comment]
We use VHR thermal imagery from UAV to extract canopy/leaf and soil/background temperatures. In addtiion the 3D cloud point might be used to characterize canopy structure and surface roughness. But the model is applied at an effective resolution that accounts for a whole canopy/soil system.

**Comment by diviningwater** [comment]
We did conduct a study in urban areas with drones and UAV data using the TSEB model (manuscript is ready for submission to a journal). The major question for us was the spatial generation of LAI. LAI is not a variable that can be downscaled (to cm/pixel), and it is limited to the sampling approach (e,g. defining an area of turfgrass like 2 * 2 or 3 * 3 m) where to collect destructive LAI samples with repetition. Therefore any LAI sample area we chose is the final LAI spatial map resolution. As a rule of thumb, the smallest ET spatial estimation (pixel size) is dictated by the coarsest input resolution (in this case LAI).
